tap_kit/client.py

```diff
@@ -31,18 +33,26 @@ def requests_method(method, request_config, body):
 
     @backoff.on_exception(backoff.expo,
                           RateLimitException,
+                          Exception,
```

Comment:
It doesn't get any broader than this; this will trigger a retry on any error resulting from your request, rendering your status checks down below as redundant. 

See my comments below for a better way to handle.